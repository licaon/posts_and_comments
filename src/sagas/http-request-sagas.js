function* startHttpRequest() {
  // This is where we add the http requests
}

export default {
  startHttpRequest,
};
