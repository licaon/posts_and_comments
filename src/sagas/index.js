import { takeEvery } from 'redux-saga';
import * as ActionTypes from 'src/constants/action-types';
import HttpRequestSagas from 'src/sagas/http-request-sagas';

function* rootSaga() {
  yield [
    takeEvery(ActionTypes.START_LOADING_DATA, HttpRequestSagas.startHttpRequest),
  ];
}

export default rootSaga;
