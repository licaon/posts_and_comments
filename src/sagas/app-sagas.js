import { put } from 'redux-saga/effects';

import { PostsActions } from 'src/actions';

export function* initApp() {
  yield put(PostsActions.getPosts());
}
