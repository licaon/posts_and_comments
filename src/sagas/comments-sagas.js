import { call, put } from 'redux-saga/effects';

import config from 'src/config';
import fetchData from 'src/api';
import { CommentsActions } from 'src/actions';

export function* getComments({ payload }) {
  const { id } = payload;
  const response = yield call(fetchData, { url: `${config.url}posts/${id}/comments` });
  yield put(CommentsActions.addComments({
    postId: id,
    comments: response,
  }));
}
