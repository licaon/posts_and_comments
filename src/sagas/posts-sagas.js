import { call, put } from 'redux-saga/effects';

import config from 'src/config';
import fetchData from 'src/api';
import { PostsActions } from 'src/actions';

export function* getPosts() {
  const response = yield call(fetchData, { url: `${config.url}posts` });
  yield put(PostsActions.addPost(response));
}

export function* deletePosts({ payload }) {
  const { id } = payload;
  const response = yield call(
    fetchData,
    { url: `${config.url}posts/${id}`, method: 'DELETE' },
  );

  if (!response.error) {
    yield put(PostsActions.confirmPostDelete({ id }));
  }
}
