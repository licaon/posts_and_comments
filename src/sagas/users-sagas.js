import { call, put } from 'redux-saga/effects';

import config from 'src/config';
import fetchData from 'src/api';
import { UsersActions } from 'src/actions';

export function* getUsers() {
  const response = yield call(fetchData, { url: `${config.url}users` });
  yield put(UsersActions.addUser(response));
}
