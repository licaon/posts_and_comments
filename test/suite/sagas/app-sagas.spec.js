import * as AppSagas from 'src/sagas/app-sagas';

describe('AppSagas', () => {
  it(`should execute initApp correctly`, (done) => {
    const generator = AppSagas.initApp();

    expect(generator.next().value).toMatchSnapshot();
    expect(generator.next().value).toMatchSnapshot();

    done();
  });
});
