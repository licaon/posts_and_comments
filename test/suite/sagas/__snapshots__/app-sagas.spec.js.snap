// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSagas should execute initApp correctly 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": Object {},
      "type": "GET_POSTS",
    },
    "channel": null,
  },
}
`;

exports[`AppSagas should execute initApp correctly 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": Object {},
      "type": "GET_USERS",
    },
    "channel": null,
  },
}
`;
